zipline/utils/events.py

```diff
@@ -605,28 +609,164 @@ def should_trigger(self, dt):
+
+        Returns
+        -------
+        rule : zipline.utils.events.EventRule
+        """
         return NthTradingDayOfMonth(n=days_offset)
 
     @staticmethod
     def month_end(days_offset=0):
+        """
+        Create a rule that triggers a fixed number of trading days before the
+        end of each month.
+
+        Parameters
+        ----------
+        days_offset : int, optional
+            Number of trading days prior to month end to trigger. Default is 0,
+            i.e., trigger on the last day of the month.
+
+        Returns
+        -------
+        rule : zipline.utils.events.EventRule
+        """
         return NDaysBeforeLastTradingDayOfMonth(n=days_offset)
 
     @staticmethod
     def week_start(days_offset=0):
+        """
+        Create a rule that triggers a fixed number of trading days after the
+        start of each week.
+
+        Parameters
+        ----------
+        days_offset : int, optional
+            Number of trading days to wait before triggering each week. Default
+            is 0, i.e., trigger on the first trading day of the week.
+        """
         return NthTradingDayOfWeek(n=days_offset)
 
     @staticmethod
     def week_end(days_offset=0):
+        """
+        Create a rule that triggers a fixed number of trading days before the
+        end of each week.
+
+        Parameters
+        ----------
+        days_offset : int, optional
+            Number of trading days prior to week end to trigger. Default is 0,
+            i.e., trigger on the last trading day of the week.
+        """
         return NDaysBeforeLastTradingDayOfWeek(n=days_offset)
 
 
 class time_rules(object):
-    market_open = AfterOpen
-    market_close = BeforeClose
+    """Factories for time-based :func:`~zipline.api.schedule_function` rules.
+
+    See Also
+    --------
+    :func:`~zipline.api.schedule_function`
+    """
+
+    @staticmethod
+    def market_open(offset=None, hours=None, minutes=None):
+        """
+        Create a rule that triggers at a fixed offset from market open.
+
+        The offset can be specified either as a :class:`datetime.timedelta`, or
+        as a number of hours and minutes.
+
+        Parameters
+        ----------
+        offset : datetime.timedelta, optional
+            If passed, the offset from market open at which to trigger. Must be
+            at least 1 minute.
+        hours : int, optional
+            If passed, number of hours to wait after market open.
+        minutes : int, optional
+            If passed, number of minutes to wait after market open.
+
+        Returns
+        -------
+        rule : zipline.utils.events.EventRule
+
+        Notes
+        -----
+        If no arguments are passed, the default offset is one minute after
+        market open.
+
+        If ``offset`` is passed, ``hours`` and ``minutes`` must not be
+        passed. Conversely, if either ``hours`` or ``minutes`` are passed,
+        ``offset`` must not be passed.
```

Comment:
Does `AfterOpen` enforce this rule programmatically?

Reply:
`_build_offset` does this.